flashflood/identifiers.py

```diff
@@ -0,0 +1,98 @@
+from uuid import uuid4
+from enum import Enum
+from functools import lru_cache
+
+from flashflood.util import datetime_from_timestamp, timestamp_now
+
+
+TOMBSTONE_SUFFIX = ".dead"
+
+
+class JournalUpdateAction(Enum):
+    UPDATE = "update"
+    DELETE = "delete"
+
+
+class JournalID(str):
+    DELIMITER = "--"
+
+    @classmethod
+    def make(cls, start_timestamp: str, end_timestamp: str, version: str, blob_id: str):
+        return cls(start_timestamp + cls.DELIMITER + end_timestamp + cls.DELIMITER + version + cls.DELIMITER + blob_id)
+
+    @classmethod
+    def from_key(cls, key):
+        return cls(key.rsplit("/", 1)[1])
```

Comment:
should there be restrictions to what form the `key` that is submitted can take?

Reply:
regex patterns could be used to constrain key format. That can be handled in future work.